exercises/concept/poetry-club-door-policy/door-policy.spec.js

```diff
@@ -7,239 +9,117 @@ import {
-    this.poem = poem;
-  }
-
-  /**
-   * Recites a poem at the door, asking the patron for a response
-   *
-   * @returns {string} key
-   */
-  recite() {
-    const patron = frontDoorPatron;
-    const key = [];
-
-    for (const line of this.poem) {
-      key.push(patron(line));
-    }
-
-    return key.join('');
-  }
-
-  assert() {
-    const password = frontDoorPassword(this.recite());
-
-    expect(password).toBe(stringify(generateCapitalized(this.poem.acrostic)));
-  }
-}
-
-class BackDoorGuard {
-  constructor(poem) {
-    this.poem = poem;
-  }
-
-  /**
-   * Recites a poem at the door, asking the patron for a response
-   *
-   * @returns {string} key
-   */
-  recite() {
-    const patron = backDoorPatron;
-    const key = [];
-
-    for (const line of this.poem) {
-      key.push(patron(line));
-    }
-
-    return key.join('');
-  }
-
-  assert() {
-    const password = backDoorPassword(this.recite());
-
-    expect(password).toBe(
-      stringify(generateCapitalized(this.poem.telestich)).concat(', please')
-    );
-  }
-}
+const recite = (poem, patron) => {
+  return poem.map((line) => patron(line));
+};
 
-const SHIRE_HORSE = new Poem(
-  `
+//Shire Horse, by Michael Lockwood
+const SHIRE_HORSE = `
 Stands so high
 Huge hooves too
 Impatiently waits for
 Reins and harness
 Eager to leave
+`
+  .trim()
+  .split('\n');
 
-Michael Lockwood
-`.trim()
-);
-
-const SHIRE_HORSE_WITH_SPACES = new Poem(
-  `
-Stands so high  
-Huge hooves too      
-Impatiently waits for 
-Reins and harness  
+//Shire Horse, by Michael Lockwood
+const SHIRE_HORSE_WITH_SPACES = `
+Stands so high
+Huge hooves too
+Impatiently waits for
+Reins and harness
 Eager to leave
+`.split('\n');
 
-Michael Lockwood
-`
-);
-
-const SUMMER = new Poem(`
+//Summer, by John Albert Caballero
+const SUMMER = `
 Sunshine warming my toes,
 Underwater fun with my friends.
 Making homemade ice cream on the porch,
 Many long nights catching fireflies.
 Early morning walks to the creek,
 Reveling in the freedom of lazy days.
+`.split('\n');
 
-John Albert Caballero
-`);
-
-const SOPHIA = new Poem(
-  `
+//Sophia, by John Albert Caballero
+const SOPHIA = `
 Serene, calming quality
 Organized, you always have it together
 Picturesque, strikingly beautiful
 Honest, so genuine
 Imaginative, a creative mind
 Alluring, so attractive
+`
+  .trim()
+  .split('\n');
 
-John Albert Caballero
-`.trim()
-);
-
-const CODE_WORK = new Poem(
-  `
+//Code Work, by Derk-Jan Karrenbeld
+const CODE_WORK = `
 Compilers intensily bestow
 On commencing without ego
 Different processes ajar
 Exit with zero quick
-
-Derk-Jan Karrenbeld
-`.trim()
-);
+`
+  .trim()
+  .split('\n');
 
 describe('strings', () => {
-  describe('front door', () => {
-    const ShireGuard = new FrontDoorGuard(SHIRE_HORSE);
-
-    test('it outputs a character per line', () => {
-      const key = ShireGuard.recite();
+  describe('front door response', () => {
+    test('should output a character per line', () => {
```

Comment:
General comment about all the test cases:  
Our test runner is not super intelligent yet so to get good output on the website, each test should include all the data it needs to run and be stand-alone. No table driven tests, no variables declared outside of the actual `test(...)` function (even if that means some duplication). You can see an example here that also has some longer data included: https://github.com/exercism/javascript/blob/304ebb8ba78283d09efce863ab104d02f7546476/exercises/concept/lasagna-master/lasagna-master.spec.js#L31-L77

Also we will connect the test cases to the tasks in the future (the one "intermediate" task the instructions currently have will be removed). So ideally there should be one `describe` block per imported function.

If you don't have time to fix those general things, I can also do it later as part of my cleanup ticket as this PR is definitely already a step in the right direction even without those fixes.

Reply:
(it's coming. WIP here: https://github.com/exercism/typescript-test-runner/pull/27)

Reply:
One describe block per function, and removed `.each` blocks.